### PR TITLE
Reconcile ImageStream only if the FG is true

### PR DESCRIPTION
**Pull request by nunnatsa**
If the enableCommonBootImageImport FG is set, create the imageStreams and reconcile them
If the enableCommonBootImageImport FG is not set, remove the imageStreams

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Reconcile the image stream only if the enableCommonBootImageImport featuregate is true
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
hco-e2e-image-index-aws lane succeeded.
/override ci/prow/hco-e2e-image-index-azure
/override ci/prow/hco-e2e-image-index-gcp

**Comment by nunnatsa** [comment]
hco-e2e-upgrade-index-azure lane succeeded.
/override ci/prow/hco-e2e-upgrade-index-aws

File: hack/check_golden_images.sh [hunk 1]
```diff
@@ -5,6 +5,12 @@ set -ex
 IMAGES_NS=${IMAGES_NS:-kubevirt-os-images}
 
 if [[ $(${KUBECTL_BINARY} get ssp -n ${INSTALLED_NAMESPACE}) ]]; then
+  [[ $(${KUBECTL_BINARY} get imageStream centos8  -n ${IMAGES_NS} --no-headers | wc -l) -eq 0 ]]
```
Comment (orenc1) [inline hack/check_golden_images.sh:8]: if you use `get` for a specific object by name, why counting the output lines and comparing to 0?
whether it's exist, or not.
you can simplify with
```
[[ ! $(${KUBECTL_BINARY} get imageStream centos8 -n ${IMAGES_NS} ]]
```

File: hack/check_golden_images.sh [hunk 2]
```diff
@@ -5,6 +5,12 @@ set -ex
...
+
+  # Test DataImportCronTemplates
+  ./hack/retry.sh 10 3 "${KUBECTL_BINARY} patch hco -n \"${INSTALLED_NAMESPACE}\" --type=json kubevirt-hyperconverged -p '[{ \"op\": \"replace\", \"path\": \"/spec/featureGates/enableCommonBootImageImport\", \"value\": true }]'"
+  sleep 10
+
   # Test image streams
   [[ $(${KUBECTL_BINARY} get imageStream centos8  -n ${IMAGES_NS} --no-headers | wc -l) -eq 1 ]]
```
Comment (orenc1) [inline hack/check_golden_images.sh:15]: same here (opposite logic)

File: pkg/controller/operands/imageStream.go [hunk 3]
```diff
@@ -32,16 +35,56 @@ var (
...
-		removeExistingOwner: false,
-		hooks:               &isHooks{required: required},
+	h := &imageStreamOperand{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "ImageStream",
+			// Previous versions used to have HCO-operator (scope namespace)
+			// as the owner of NetworkAddons (scope cluster).
+			// It's not legal, so remove that.
```
Comment (orenc1) [inline pkg/controller/operands/imageStream.go:84]: this comment doesn't seem relevant here

**Review by orenc1 — COMMENTED** [review]
looks good so far, haven't went over all of the changes yet.
some nits inline

**Pull request merged**